rust/migration/src/m20231213_194935_create_blob_table.rs

```diff
@@ -0,0 +1,141 @@
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .create_table(
+                Table::create()
+                    .table(BlobStore::Table)
+                    .col(
+                        ColumnDef::new(BlobStore::Id)
+                            .integer()
+                            .not_null()
+                            .auto_increment()
+                            .primary_key(),
+                    )
+                    .col(ColumnDef::new(BlobStore::Name).string().not_null())
+                    .to_owned(),
+            )
+            .await?;
+
+        // Insert basic backing stores since they must always exist.
+        let insert = Query::insert()
+            .into_table(BlobStore::Table)
+            .columns([BlobStore::Id, BlobStore::Name])
+            .values_panic([1.into(), "local".into()])
+            .values_panic([2.into(), "artifactory".into()])
+            .values_panic([3.into(), "aws".into()])
+            .values_panic([4.into(), "gcs".into()])
+            .values_panic([5.into(), "azure".into()])
+            .to_owned();
+
+        manager.exec_stmt(insert).await?;
+
+        manager
+            .create_table(
+                Table::create()
+                    .table(Blob::Table)
+                    .col(ColumnDef::new(Blob::Id).string().not_null().primary_key())
+                    .col(ColumnDef::new(Blob::Store).integer().not_null())
+                    .col(ColumnDef::new(Blob::WorkspacePath).string().not_null())
+                    .foreign_key(
+                        ForeignKeyCreateStatement::new()
+                            .name("fk-store-blob_store")
+                            .from_tbl(Blob::Table)
+                            .from_col(Blob::Store)
+                            .to_tbl(BlobStore::Table)
+                            .to_col(BlobStore::Id)
+                            // All blobs for a given store must be manually
+                            // deleted before the blob store may be deleted.
+                            .on_delete(ForeignKeyAction::Restrict),
+                    )
+                    .to_owned(),
+            )
+            .await?;
+
+        manager
```

Comment:
You're creating a new table here but I don't think that's what you want, we already have a table that stores output files for this purpose.